internal/backend/remote-state/vault/backend.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"net"
+	"time"
+
+	vaultapi "github.com/hashicorp/vault/api"
+	"github.com/opentofu/opentofu/internal/backend"
+	"github.com/opentofu/opentofu/internal/encryption"
+	"github.com/opentofu/opentofu/internal/legacy/helper/schema"
+)
+
+// New creates a new backend for Vault remote state.
+func New(enc encryption.StateEncryption) backend.Backend {
+	s := &schema.Backend{
+		Schema: map[string]*schema.Schema{
+			"mount": &schema.Schema{
```

Comment:
Will align these attributes based on the vault provider and vault_kv2_secret resource